fe/fe-core/src/main/java/org/apache/doris/analysis/ShowPartitionsStmt.java

```diff
@@ -137,8 +151,14 @@ public void analyze(Analyzer analyzer) throws UserException {
     public void analyzeImpl(Analyzer analyzer) throws UserException {
         super.analyze(analyzer);
         tableName.analyze(analyzer);
-        // disallow external catalog
-        Util.prohibitExternalCatalog(tableName.getCtl(), this.getClass().getSimpleName());
+        // disallow unsupported catalog
+        String catalogName = tableName.getCtl();
+        catalog = Env.getCurrentEnv().getCatalogMgr().getCatalog(catalogName);
```

Comment:
catalog may be null here if `catalogName` does not exists.

Reply:
fixed